### PR TITLE
chore: [theia-devworkspace handler] store projects section when generating templates

### PR DESCRIPTION
**Pull request by vinokurig**
Signed-off-by: Igor Vinokur <ivinokur@redhat.com>

### What does this PR do?
Add an optional parameter `--project.<project name>=<project location>`
If defined, project section in a zip format is added to the devfile template:
```yaml
projects:
    - name: project-name
      zip:
        location: '{{DEVFILE_REGISTRY_URL}}/project-name.zip'
```
If not defined, project section in a git format is added to the devfile template:
```yaml
projects:
  - name: java-spring-petclinic
    git:
      remotes:
        origin: https://github.com/che-samples/java-spring-petclinic.git
      checkoutFrom:
        revision: devfilev2
```
### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/20877

### How to test this PR?

1. Build the `devworkspace-handler` package
2. Run `node lib/entrypoint.js --devfile-url:https://github.com/che-samples/java-spring-petclinic/tree/devfilev2 --output-file:/Users/ivinokur/projects/che-theia/tools/devworkspace-handler/a.yaml` in the package root and see the default projects section in the generated template.
3. Run `node lib/entrypoint.js --devfile-url:https://github.com/che-samples/java-spring-petclinic/tree/devfilev2 --output-file:/Users/ivinokur/projects/che-theia/tools/devworkspace-handler/a.yaml --project.name={{DEVFILE_REGISTRY_URL}}/project.zip` and see the projects section with the specified projects in the zip format.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [ ] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [ ] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [ ] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)
- [ ] Optional Companion PR for updating the HappyPath tests is approved and ready to be merged

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### Happy Path Channel

HAPPY_PATH_CHANNEL=stable

### CODE REVIEW

File: tools/devworkspace-handler/src/generate.ts [hunk 1]
```diff
@@ -53,6 +54,20 @@ export class Generate {
     // devfile of the editor
     const editorDevfile = await this.pluginRegistryResolver.loadDevfilePlugin(editorEntry);
 
+    editorDevfile.projects = [];
+    if (projects.length > 0) {
+      projects.forEach(p => {
+        editorDevfile.projects.push({ name: p.name, zip: { location: p.location } });
```
Comment (benoitf) [inline tools/devworkspace-handler/src/generate.ts:60]:
```suggestion
        editorDevfile.projects = projects.map(project => ({ name: project.name, zip: { location: project.location } }));
```

File: tools/devworkspace-handler/src/generate.ts [hunk 2]
```diff
@@ -53,6 +54,20 @@ export class Generate {
...
+      });
```
Comment (benoitf) [inline tools/devworkspace-handler/src/generate.ts:61]: editorDevfile.projects = projects.map(project => ({ name: project.name, zip: { location: project.location }

**Review by benoitf — APPROVED** [review]
It might be good if the --projects could override location of existing project name rather than providing a complete different list

**Review by svor — APPROVED** [review]

**Comment by vinokurig** [comment]
Since the devworkspce deployment is broken on minikube https://github.com/eclipse/che/issues/20888 I want to postpone the merge until the issue is fixed to be able to test the PR.

**Comment by benoitf** [comment]
I think it's safe to be merged as we're only picking up specific versions on consumer side

**Pull request merged**